### PR TITLE
Topic/an 698 - Adding mode and format to Destination reference

### PR DESCRIPTION
**Pull request by sraj-sm**

### CODE REVIEW

File: client/models.go [hunk 1]
```diff
@@ -281,12 +285,14 @@ type GCSStagingArea struct {
...
-	FolderPartitioningEnabled *bool  `json:"folderPartitioningEnabled,omitempty"`
-	TableName                 string `json:"tableName,omitempty"`
-	Topic                     string `json:"topic,omitempty"`
+	Type                      string      `json:"adt_type"`
+	DestinationID             int         `json:"destinationId"`
+	Folder                    string      `json:"folder,omitempty"`
+	FolderPartitioningEnabled *bool       `json:"folderPartitioningEnabled,omitempty"`
+	TableName                 string      `json:"tableName,omitempty"`
+	Topic                     string      `json:"topic,omitempty"`
+	Format                    KafkaFormat `json:"format,omitempty"`
```
Comment (HuwCampbell) [inline client/models.go:294]: Should be a pointer.

File: client/resource_feature_store.go [hunk 2]
```diff
@@ -169,6 +169,11 @@ func folderDestinationSchema() *schema.Resource {
 				Type:     schema.TypeBool,
 				Required: true,
 			},
+			"save_mode": {
+            		    Type:     schema.TypeString,
+            		    Required: true,
+            		    ValidateFunc: validation.StringIsNotWhiteSpace,
```
Comment (HuwCampbell) [inline client/resource_feature_store.go:175]: This should validate that it's in a slice of correct values. See aggregation type in feature.

File: client/resource_feature_store.go [hunk 3]
```diff
@@ -193,6 +203,11 @@ func topicDestinationSchema() *schema.Resource {
 				Required:     true,
 				ValidateFunc: validation.StringIsNotWhiteSpace,
 			},
+			"format": {
+            			Type:     schema.TypeString,
+            			Required: true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+            		},
```
Comment (HuwCampbell) [inline client/resource_feature_store.go:210]: Formatting and validation. VScode should get this right on save, but gofmt should be run here.

File: client/models.go (hunk 1 above)
Comment (sraj-sm) [reply client/models.go]: done

File: client/resource_feature_store.go (hunk 2 above)
Comment (sraj-sm) [reply client/resource_feature_store.go]: done

File: client/resource_feature_store.go (hunk 3 above)
Comment (sraj-sm) [reply client/resource_feature_store.go]: done

**Pull request merged**